dev/PipsControl/PipsControl.cpp

```diff
@@ -0,0 +1,335 @@
+    const auto templateSettings = winrt::make<PipsControlTemplateSettings>();
+    templateSettings.SetValue(PipsControlTemplateSettings::s_VerticalPipsItemsProperty, m_verticalPipsElements);
+    SetValue(s_TemplateSettingsProperty, templateSettings);
+
+    SetDefaultStyleKey(this);
+
+}
+
+PipsControl::~PipsControl() {
+    m_nextPageButtonClickRevoker.revoke();
+    m_previousPageButtonClickRevoker.revoke();
+    m_verticalPipsElementPreparedRevoker.revoke();
+    m_rootGridPointerEnteredRevoker.revoke();
+    m_rootGridPointerExitedRevoker.revoke();
+    m_rootGridKeyDownRevoker.revoke();
+}
+
+void PipsControl::OnApplyTemplate()
+{
+    const winrt::IControlProtected controlProtected = *this;
+    m_previousPageButtonClickRevoker.revoke();
+
+    [this](const winrt::Button button) {
+        if (button) {
+            winrt::AutomationProperties::SetName(button, ResourceAccessor::GetLocalizedStringResource(SR_PipsControlPreviousPageButtonTextName));
+            m_previousPageButtonClickRevoker = button.Click(winrt::auto_revoke, { this, &PipsControl::OnPreviousButtonClicked });
+        }
+    }(GetTemplateChildT<winrt::Button>(c_previousPageButtonName, *this));
+
+    m_nextPageButtonClickRevoker.revoke();
+    [this](const winrt::Button button) {
+        if (button) {
+            winrt::AutomationProperties::SetName(button, ResourceAccessor::GetLocalizedStringResource(SR_PipsControlNextPageButtonTextName));
+            m_nextPageButtonClickRevoker = button.Click(winrt::auto_revoke, { this, &PipsControl::OnNextButtonClicked });
+        }
+    }(GetTemplateChildT<winrt::Button>(c_nextPageButtonName, *this));
+
+    m_verticalPipsElementPreparedRevoker.revoke();
+    [this](const winrt::ItemsRepeater repeater) {
+        m_verticalPipsRepeater.set(repeater);
+        if (repeater) {
+            m_verticalPipsElementPreparedRevoker = repeater.ElementPrepared(winrt::auto_revoke, { this, &PipsControl::OnElementPrepared });
+        }
+    }(GetTemplateChildT<winrt::ItemsRepeater>(c_PipsControlPipsRepeaterName, *this));
+
+
+    m_verticalPipsScrollViewer.set(GetTemplateChildT<winrt::FxScrollViewer>(c_PipsControlScrollViewerName, *this));
+
+    m_rootGridPointerEnteredRevoker.revoke();
+    m_rootGridPointerExitedRevoker.revoke();
+
+    [this](const winrt::Grid grid) {
+        if (grid) {
+            m_rootGridPointerEnteredRevoker = grid.PointerEntered(winrt::auto_revoke, { this, &PipsControl::OnPipsControlPointerEntered });
+            m_rootGridPointerExitedRevoker = grid.PointerExited(winrt::auto_revoke, { this, &PipsControl::OnPipsControlPointerExited });
+            m_rootGridKeyDownRevoker = grid.KeyDown(winrt::auto_revoke, { this, &PipsControl::OnRootGridKeyDown });
+
+        }
+    }(GetTemplateChildT<winrt::Grid>(c_rootGridName, controlProtected));
+
+    OnNumberOfPagesChanged(0);
+    OnSelectedPageIndexChange(-1);
+    OnMaxDisplayedPagesChanged(0);
```

Comment:
Do we need these ?

Reply:
@ranjeshj yup, to update the items array and set vertical pips size as well as buttons visibility